packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -30,14 +32,13 @@ interface ComboboxOption {
   description?: string;
 }
 
-interface ComboboxProps
-  extends React.InputHTMLAttributes<
-    Omit<HTMLInputElement, 'value' | 'defaultValue'>
-  > {
+type ComboboxProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  'value' | 'defaultValue'
+> & {
   label: ContentNode;
   options?: ComboboxOption[];
-  value?: ComboboxValue;
-  defaultValue?: ComboboxValue;
+  inputValue?: string;
```

Comment:
I apologize because this wasn't well defined, but I'm not sure we should be changing `value` to `inputValue` here. Existing users of the component are expecting to use `value` for non-multiselect components so we need to maintain that compatibility in order to not be a breaking change.

What we may need is a mechanism for converting a value string into a multiselect value. Feel free to update the RFC as we need to reach a consensus to ensure the changes we make here are compatible and make sense.

Reply:
I had no idea how to convert value into multiselect value, therefore added this prop. It's just not compatible - `onChange` emits `string` and `onSelectionChange` emits `string[]` values. One of the possible solutions - albeit radical one - would be only left `value` and `onSelectionChange` props, and remove `onChange`. Although I don't know if that's plausible for users of the component.

Reply:
@scurker what would you think about creating a new `MultiSelectCombobox` component? I don't totally love the idea (feels a little lazy), but I also don't hate it. Thoughts?

Reply:
> @scurker what would you think about creating a new MultiSelectCombobox component? I don't totally love the idea (feels a little lazy), but I also don't hate it. Thoughts?

There's a lot of overlap between the single/multiselect components, so that introduces an additional challenge of keeping those components in sync if we split them up with future updates. It's also a higher lift for authors since swapping between a single/multiselect component is potentially as small as changing the property with the current proposal.

Reply:
> I had no idea how to convert value into multiselect value, therefore added this prop. It's just not compatible - onChange emits string and onSelectionChange emits string[] values.

That's where the discriminated union could potentially help, example below.

```tsx
type ComboboxProps = {
  ...
  onChange: (value: string, ...) => {} 
}

type MultiselectComboboxProps extends Omit<ComboboxProps, 'onChange'> = {
  ...
  onChange: (value: string[]) => {}
}
```